src/main/scala/firrtl/AddDescriptionNodes.scala

```diff
@@ -28,6 +28,13 @@ case class DocStringAnnotation(target: Target, description: String) extends Desc
       case Some(seq) => seq.map(n => this.copy(target = n))
     }
   }
+  override private[firrtl] def dedup: Option[(Any, Annotation, ReferenceTarget)] = this match {
+    case a @ DocStringAnnotation(refTarget: ReferenceTarget, _) =>
+      Some(((refTarget.pathlessTarget, description), copy(target = refTarget.pathlessTarget), refTarget))
+    case a @ DocStringAnnotation(pathTarget: InstanceTarget, _) =>
+      Some(((pathTarget.pathlessTarget, description), copy(target = pathTarget.pathlessTarget), pathTarget.asReference))
```

Comment:
```suggestion
    case a @ DocStringAnnotation(target: IsComponent, _) =>
      Some(((target.pathlessTarget, description), copy(target = target.pathlessTarget), target))
```
I think we can simplify and just use `IsComponent` which is a trait mixed in by `ReferenceTarget` and `InstanceTarget`. The one complication is that the declaration of this method is currently:
```scala
override private[firrtl] def dedup: Option[(Any, Annotation, ReferenceTarget)] 
```
Why is that last part of the tuple a `ReferenceTarget`? Maybe that type should change.

Reply:
Future note: `dedup`'s return type will change in a later PR which will improve the annotation deduplication algorithm in general.